lib/sales_engine.rb

```diff
@@ -12,25 +12,28 @@ def initialize(csv_data)
     routes(csv_data)
   end
 
-  def top_day_of_the_week
-    # invoice_hash = @invoices.all.group_by do |invoice|
-    #   invoice.created_at
-    # end.flatten
-    # invoice_hash
-    # require "pry"; binding.pry
-
-    created_at_array = []
-    popular_days = []
-
-    @invoices.all.each do |invoice|
-        created_at_array << invoice.created_at
+  def find_invoice_status_percentage(status)
+    matched_status = @invoices.all.find_all do |invoice|
+      invoice.status == status
     end
+    percentage = ((matched_status.count.to_f / @invoices.all.count.to_f) * 100)
+    percentage.round(2)
+  end
 
-    this_is_the_most_popular_day =  created_at_array.max_by{|day|created_at_array.count(day)}
-    # require "pry"; binding.pry
-        # popular_days << this_is_the_most_popular_day
-        # created_at_array.delete(this_is_the_most_popular_day)
-
+  def top_day_of_the_week
+    invoice_count = Hash.new(0)
```

Comment:
This is a neat way to get both the hash datatype we want to work with, and also have our initial value for `invoice` to be set at `0`, that way with each iteration it can keep adding how many times the occurrence of the key `invoice.created_at` is happening. Nice work on this here!